File: prime-number-checker/prime-number.cpp
#include <iostream>
#include <cmath>

using namespace std;

// ini fungsi global
bool isPrime(int number)
{
  if (number < 2)
  {
    return false;
  }

  for (int i = 2; i <= sqrt(number); i++)
  {
    if (number % i == 0)
    {
      return false;
    }
  }

  return true;
}

// ini fungsi utama
int main()
{
  while (true)
  {
    int number;
    cout << "Enter for prime number test (enter 0 to exit): ";
    cin >> number;

    if (number == 0)
    {
      break;
    }

    if (isPrime(number))
    {
      cout << number << " is a prime number." << endl;
    }
    else
    {
      cout << number << " is not a prime number" << endl;
    }
  }

  return 0;
}

File: prime-number-checker/prime-number.py
def is_prime(number):
    if number < 2:
        return False

    for i in range(2, int(number**0.5) + 1):
        if number % i == 0:
            return False

    return True

while True:
    number = int(input("Enter the number for prime number test (enter 0 to exit): "))
    
    if number == 0:
        break

    if is_prime(number):
        print(f"{number} is a prime number.")
    else:
        print(f"{number} is not a prime number.")
